### PR TITLE
Fix shared element animation for views with translation style

### LINKED ISSUE #6729: Shared Element Transitions: Animating to transformed view doesn't work correctly
### ⚠️  First off: I think this is related to #6728. 

When animating Shared Element Transitions to a view that is transformed in some sort (`translateX` in my case, but might also be `scale`, `translateY`, ...), the Shared Element Transitions weirdly miscalculates the destination coordinates at first, but manages to get it right halfway through. Yes, that's my explanation. It's weird, take a look at it yourself:

📹  Here's a video demo: https://i.imgur.com/tNcQwD1.mp4

So in this example, I'm translating the parent view across the X axis, so first Post is `translateX: 0` which works perfectly fine. Second post is `translateX: -SCREEN_WIDTH`, which already looks a bit off. And then it just goes on, with the animation going further to the origin point (`0` aka "left") the more you translate x.

## To Reproduce

1. Create parent view that translates X axis
2. Create list of views in that translated view (Shared Elements)
3. Push a Modal with SETs that animates the Shared Element (image)
4. Dismiss the modal with the SET

## Expected behavior

I expect the animation to work normally

## Actual Behavior

The animation swiftly goes towards the origin (`0` aka "left"), then finds out it's a bit drunk and finds it's way back home (actual position)

## Your Environment

- React Native Navigation version: master (snapshot)
- React Native version: 0.63.3
- Platform(s) (iOS, Android, or both?): iOS

## Reproducible Demo

@yogevbd hit me up on discord for a repro demo

## Related Issues

* SET for Modals: #6647
* #6728

### PR DESCRIPTION
**Pull request by yogevbd**
Closes #6729

### CODE REVIEW

**Review by mrousavy — COMMENTED** [review]
I didn't notice any difference between this, and current master (https://github.com/wix/react-native-navigation/commit/6835d63d392c889d7c36fa080aaec9dfb253c731)? 🤨

I tested the exact interaction shown in https://github.com/wix/react-native-navigation/issues/6729, as well as an image viewer opening which previously wildly rotated 3 times on inverted FlatLists 🤔

**Comment by yogevbd** [comment]
@mrousavy could you please verify that it solves the translate and rotation issues after latest changes?

**Comment by mrousavy** [comment]
@yogevbd just tested again, and I didn't notice any difference between the new commits and before. (Before = just the first commit of this PR).

By "rotation issue" you mean the transition in my chat which weirdly spinned around the Y axis? I don't have that issue anymore, that was already fixed with https://github.com/wix/react-native-navigation/commit/6835d63d392c889d7c36fa080aaec9dfb253c731!

So to recap, with this PR, the two things that don't work are:
1. Different `resizeMode`s (doesn't work on master either, not yet implemented)
2. The zIndex of the overlapping views in the story animation (does actually work on master, DM'd you a video demo on discord)

**Comment by yogevbd** [comment]
@mrousavy let's open another issue for the `resizeMode` as it isn't related to this PR and will require some heavy work..
The zindex issue solved in master so maybe your local branch isn't up to date with the latest merge from master.
This PR should fix transitions for views with `transform` styles (`translateX, translateY, rotate` etc).

**Review by hafixo — APPROVED** [review]

**Review by hafixo — APPROVED** [review]
Merge pull requested #6796
